Open Judge System/Web/OJS.Web/Areas/Contests/Controllers/CompeteController.cs

```diff
@@ -174,23 +204,29 @@ public ActionResult Register(int id, bool official)
             if (participantFound)
             {
                 // Participant exists. Redirect to index page.
-                return this.RedirectToAction(GlobalConstants.Index, new { id, official });
+                return this.RedirectToAction(GlobalConstants.Index, new {id, official});
             }
 
             var contest = this.Data.Contests.All().Include(x => x.Questions).FirstOrDefault(x => x.Id == id);
 
-            this.ValidateContest(contest, official);
+            try
+            {
+                this.ValidateContest(contest, official);
+                
+                if (contest.ShouldShowRegistrationForm(official))
```

Comment:
Do we really need this check, if we're already on the Page for registering to the contest?

Reply:
Yes, I modified the logic and the adding of Participant is happeninig only in the get method of register or in the post if the contest requires password